test/module/irohad/ametsuchi/ametsuchi_test.cpp

```diff
@@ -256,79 +266,45 @@ TEST_F(AmetsuchiTest, queryGetAccountAssetTransactionsTest) {
   auto wsv = storage->getWsvQuery();
   auto blocks = storage->getBlockQuery();
 
-  const auto admin = "admin1";
-  const auto domain = "domain";
-  const auto user1name = "user1";
-  const auto user2name = "user2";
-  const auto user3name = "user3";
-  const auto user1id = "user1@domain";
-  const auto user2id = "user2@domain";
-  const auto user3id = "user3@domain";
-  const auto asset1name = "asset1";
-  const auto asset2name = "asset2";
-  const auto asset1id = "asset1#domain";
-  const auto asset2id = "asset2#domain";
+  const auto admin = "admin1", domain = "domain", user1name = "user1",
+             user2name = "user2", user3name = "user3", user1id = "user1@domain",
+             user2id = "user2@domain", user3id = "user3@domain",
+             asset1name = "asset1", asset2name = "asset2",
+             asset1id = "asset1#domain", asset2id = "asset2#domain";
 
-  CreateRole createRole;
-  createRole.role_name = "user";
-  createRole.permissions = {can_add_peer, can_create_asset, can_get_my_account};
+  std::string account, src_account, dest_account, asset;
+  iroha::Amount amount;
 
   // 1st tx
   Transaction txn;
   txn.creator_account_id = admin;
 
   // Create domain
-  CreateDomain createDomain;
-  txn.commands.push_back(std::make_shared<CreateRole>(createRole));
-  createDomain.domain_id = domain;
-  createDomain.user_default_role = "user";
-  txn.commands.push_back(std::make_shared<CreateDomain>(createDomain));
+  txn.commands.push_back(std::make_shared<CreateRole>(
+      "user",
+      std::set<std::string>{
+          can_add_peer, can_create_asset, can_get_my_account}));
+  txn.commands.push_back(cmd_gen.generateCreateDomain(domain, "user"));
+
+  // Create accounts
+  for (const auto &name : {user1name, user2name, user3name}) {
```

Comment:
Accumulate will require storing this initializer list in a separate variable, since template cannot deduce type of brace-init list directly, which will make code more complex.